python-sdk/tutorials/automl-with-azureml/forecasting-pipelines/auto-ml-forecasting-pipelines.ipynb

```diff
@@ -0,0 +1,783 @@
+    "\n",
+    "#### Create or Attach existing AmlCompute\n",
+    "\n",
+    "You will need to create a compute target for your AutoML run. In this tutorial, you create AmlCompute as your training compute resource.\n",
+    "\n",
+    "> Note that if you have an AzureML Data Scientist role, you will not have permission to create compute resources. Talk to your workspace or IT admin to create the compute targets described in this section, if they do not already exist.\n",
+    "\n",
+    "#### Creation of AmlCompute takes approximately 5 minutes. \n",
+    "If the AmlCompute with that name is already in your workspace this code will skip the creation process.\n",
+    "As with other Azure services, there are limits on certain resources (e.g. AmlCompute) associated with the Azure Machine Learning service. Please read [this article](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-manage-quotas) on the default limits and how to request more quota."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.compute import ComputeTarget, AmlCompute\n",
+    "from azureml.core.compute_target import ComputeTargetException\n",
+    "\n",
+    "# Choose a name for your CPU cluster\n",
+    "amlcompute_cluster_name = \"forecast-step-cluster\"\n",
+    "\n",
+    "# Verify that cluster does not exist already\n",
+    "try:\n",
+    "    compute_target = ComputeTarget(workspace=ws, name=amlcompute_cluster_name)\n",
+    "    print(\"Found existing cluster, use it.\")\n",
+    "except ComputeTargetException:\n",
+    "    compute_config = AmlCompute.provisioning_configuration(\n",
+    "        vm_size=\"STANDARD_DS12_V2\", max_nodes=4\n",
+    "    )\n",
+    "    compute_target = ComputeTarget.create(ws, amlcompute_cluster_name, compute_config)\n",
+    "compute_target.wait_for_completion(show_output=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run Configuration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azureml.core.runconfig import CondaDependencies, RunConfiguration\n",
+    "\n",
+    "# create a new RunConfig object\n",
+    "conda_run_config = RunConfiguration(framework=\"python\")\n",
+    "\n",
+    "# Set compute target to AmlCompute\n",
+    "conda_run_config.target = compute_target\n",
+    "\n",
+    "conda_run_config.docker.use_docker = True\n",
+    "\n",
+    "cd = CondaDependencies.create(\n",
+    "    pip_packages=[\n",
+    "        \"azureml-sdk[automl]\",\n",
+    "        \"applicationinsights\",\n",
+    "        \"azureml-opendatasets\",\n",
+    "        \"azureml-defaults\",\n",
+    "    ],\n",
+    "    conda_packages=[\"numpy==1.16.2\"],\n",
+    "    pin_sdk_version=False,\n",
+    ")\n",
+    "conda_run_config.environment.python.conda_dependencies = cd\n",
```

Comment:
Could we use curated environment here? May be we can even create the helper function (with the perspective of moving it to SDK eventually)
```
from azureml.train.automl.constants import Scenarios
from azureml._restclient.jasmine_client import JasmineClient

jasmine_client = JasmineClient(
        service_context=experiment.workspace.service_context,
        experiment_name=experiment.name,
        experiment_id=experiment.id,
    )
predict_env = jasmine_client.get_curated_environment(
    scenario=Scenarios.AUTOML,
    enable_dnn=False,
    enable_gpu=False,
    compute=amlcompute_cluster_name,
    compute_sku=amlcompute_cluster_name.vm_size,
)
```

Reply:
This is the environment for the `register` which does not actually need all the dependencies for AutoML.